lib/Frontend/DiagnosticVerifier.cpp

```diff
@@ -601,10 +651,23 @@ DiagnosticVerifier::Result DiagnosticVerifier::verifyFile(unsigned BufferID) {
     for (; Count; --Count)
       ExpectedDiagnostics.push_back(Expected);
   }
+}
 
-  
-  // Make sure all the expected diagnostics appeared.
-  std::reverse(ExpectedDiagnostics.begin(), ExpectedDiagnostics.end());
+void DiagnosticVerifier::verifyAllExpectedDiagnosticsAppeared(
+    StringRef BufferName, StringRef InputFile,
+    std::vector<ExpectedDiagnosticInfo> &ExpectedDiagnostics,
+    std::vector<llvm::SMDiagnostic> &Errors) {
+  using llvm::SMLoc;
+
+  auto addError = [&](const char *Loc, const Twine &message,
```

Comment:
Can the various `addError` lambdas be a single, shared function or method? It looks like the only difference is which boolean you're testing.

Reply:
Hmm... I think I have something I can do... Let me see.